src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,141 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (isEmpty()) {
+            head = tail = newNode;
+        }
+        tail.next = newNode;
```

Comment:
The `add` method should include an `else` block for setting `tail.next = newNode;` to ensure it only executes when the list is not empty. Otherwise, this line will throw a `NullPointerException` when adding the first element.